content/docs/refs-and-the-dom.md

```diff
@@ -29,32 +31,69 @@ For example, instead of exposing `open()` and `close()` methods on a `Dialog` co
 
 Your first inclination may be to use refs to "make things happen" in your app. If this is the case, take a moment and think more critically about where state should be owned in the component hierarchy. Often, it becomes clear that the proper place to "own" that state is at a higher level in the hierarchy. See the [Lifting State Up](/docs/lifting-state-up.html) guide for examples of this.
 
-### Adding a Ref to a DOM Element
+>**Note:**
+>
+>The examples below have updated to use the `React.createRef()` API introduced in React 16.3.
+
+### Creating Refs
 
-React supports a special attribute that you can attach to any component. The `ref` attribute takes a callback function, and the callback will be executed immediately after the component is mounted or unmounted.
+Refs are created using `React.createRef()` and attached to React elements via the `ref` attribute. Refs are commonly assigned to an instance property when a component is constructed so they can be referenced throughout the the component.
 
-When the `ref` attribute is used on an HTML element, the `ref` callback receives the underlying DOM element as its argument. For example, this code uses the `ref` callback to store a reference to a DOM node:
+```javascript{4,7}
+class MyComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.myRef = React.createRef();
+  }
+  render() {
+    return <div ref={this.myRef} />;
+  }
+}
+```
+
+### Accessing Refs
+
+When a ref is passed to an element in `render`, a reference to the node becomes accessible at the `value` attribute of the ref.
```

Comment:
Wording nit: We have a lot of sentences with both "ref" and "reference". It's fine, but I wonder if it might read better if we swapped "reference" with "pointer" in some of these places?

> When a ref is passed to an element in `render`, a pointer to the node becomes accessible at the `value` attribute of the ref.

Reply:
Maybe... but pointer isn't particularly common terminology in JS-land so I'd worry it might confuse some people.

Reply:
Terms like "pointer" and "reference" aren't just used in programming. I think they have broader meaning outside of programming that's relevant.

Maybe someone like @gaearon could weigh-in with an opinion about non-native English speakers and if there are concerns there though. I have no idea.